src/charm.py

```diff
@@ -699,7 +719,12 @@ def _process_ingresses(self, ingresses):
                 # Ensure that the exact same route for this route was not yet defined.
                 defined_paths = ingress_paths[rule.host]
                 for path in rule.http.paths:
-                    duplicate_paths = list(filter(lambda p: p.path == path.path, defined_paths))
+                    duplicate_paths = list(
+                        filter(
+                            lambda p: p.path == path.path,  # pylint: disable=cell-var-from-loop
```

Comment:
SHould we note why this disable is reasonable in this case please?

Reply:
This can also be done without a lambda: `duplicate_paths = [defined_path for defined_path in defined_paths if defined_path.path == path.path]`

Writing this I'm confused by all the paths, the variable names here could be improved for both `defined_paths` and `path` since both of them have a `path` attribute which means that the outer variable probably is more than just a `path`

Reply:
+1 for rephrasing, all this "path path, path" is rhythmic, though it's better suited in a song and not in code.

Reply:
Changed.